src/middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { NextRequest, NextResponse } from "next/server";
+import { client } from "../studio/lib/client";
+import { RedirectSparsePage } from "../studio/lib/payloads/redirect";
+
+export async function middleware(request: NextRequest) {
+  const slug = request.nextUrl.pathname;
+  const redirect = await client.fetch<RedirectSparsePage | null>(
+    `*[_type == "redirect" && source.current == "${slug}"][0]{
```

Comment:
I like setting queries as const for better readability, not sure it's needed here, but then I know what the query asks for and returns